src/components/layout/Navbar.tsx

```diff
@@ -0,0 +1,171 @@
+
+  interface LinkProps {
+    name: string;
+    buttonVariant: string;
+    link: string;
+    color?: string;
+  }
+
+  const navbarContainerStyles: SxProps = {
+    ...defaultFlexRowStyles,
+    px: 1.5,
+    py: 1,
+    justifyContent: "space-between",
+    height: "4rem",
+  };
+
+  const imageStyles: SxProps = {
+    height: "100%",
+    objectFit: "contain",
+  };
+
+  const pages: LinkProps[] = [
+    {
+      name: "home",
+      buttonVariant: "text",
+      link: "/",
+    },
+    {
+      name: "about",
+      buttonVariant: "text",
+      link: "/",
+    },
+    {
+      name: "what we do",
+      buttonVariant: "text",
+      link: "/",
+    },
+    {
+      name: "join us",
+      buttonVariant: "contained",
+      link: "/",
+      color: "secondary",
+    },
+  ];
+
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const open = Boolean(anchorEl);
+  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const mobileNavLinks = (
+    <Box sx={displayOnMobileStyles}>
+      <Button
+        id="demo-positioned-button"
+        aria-controls={open ? "demo-positioned-menu" : undefined}
+        aria-haspopup="true"
+        aria-expanded={open ? "true" : undefined}
+        onClick={handleClick}
+      >
+        <FontAwesomeIcon icon={faBars} />
+      </Button>
+      <Menu
+        id="demo-positioned-menu"
+        aria-labelledby="demo-positioned-button"
+        anchorEl={anchorEl}
+        open={open}
+        onClose={handleClose}
+        anchorOrigin={{
+          vertical: "top",
+          horizontal: "left",
+        }}
+        transformOrigin={{
+          vertical: "top",
+          horizontal: "left",
+        }}
+      >
+        {pages.map(({ name, buttonVariant, link, color }) =>
+          color ? (
+            <MenuItem>
+              {
+                // @ts-expect-error there is a known bug with MUI types where a string can be used by TypeScript is not
+                // able to parse the autmoatic type conversion
+                <Button href={link} variant={buttonVariant} color={color}>
+                  {name}
+                </Button>
+              }
+            </MenuItem>
+          ) : (
+            <MenuItem>
+              {
+                // @ts-expect-error there is a known bug with MUI types where a string can be used by TypeScript is not
+                // able to parse the autmoatic type conversion
+                <Button href={link} variant={buttonVariant}>
+                  {name}
+                </Button>
+              }
+            </MenuItem>
+          )
+        )}
+      </Menu>
+    </Box>
+  );
+
+  const desktopNavLinks = (
+    <Box sx={{ ...defaultFlexRowStyles, display: { md: "flex", xs: "none" } }}>
+      {pages.map(({ name, buttonVariant, link, color }) =>
```

Comment:
I can't actually because one of them only requires a `<Button>` component and the other one requires that to be wrapped in a `<MenuItem>` component due to its usage in a dropdown.